src/H5FDs3comms.c

```diff
@@ -136,12 +152,21 @@ H5FD__s3comms_curl_write_callback(char *ptr, size_t size, size_t nmemb, void *us
     struct s3r_datastruct *sds    = (struct s3r_datastruct *)userdata;
     size_t                 nbytes = size * nmemb;
 
-    /* Write bytes and size to userdata/sds struct */
-    if (size > 0) {
```

Comment:
`size` will always be 1 here, so changed this to check the total data size instead.